hw3/03-form.js

```diff
@@ -0,0 +1,63 @@
+const http = require('http');
+const URLSearchParams = require('url-search-params');
+
+const port = process.env.PORT || 5001;
+
+// http://localhost:5001/form should return a form with input elements for username, email, and submit button
+
+// http://localhost:5001/submit should return all the data the user entered
+
+// HTML form to be used for route /form
+const postHTML = `<html><head></head><body>
+  <form method='post' accept-charset=utf-8>
+  <label for="name">Name: </label>
+  <input type="text" name="name" id="name"><br />
+  <label for="email">Email: </label>
+  <input type="text" name="email" id="email"><br />
+  <label for="textInput">Submit your message</label>
+  <textarea type="text" name="textInput"id="textInput"></textarea><br />
+  <label for=newsLetterInput>Sign up for the newsletter</label>
+  <input type=checkbox name="newsLetterInpu" id=newsLetterInput><br />
+  <input type='submit'>
+  </form></body></html>`;
+
+let body = '';
+const server = http.createServer((req, res) => {
+  if (req.url === '/form') {
+    res.writeHead(200, { 'Content-Type': 'text/html' });
+    req.on('data', (chunk) => {
+      body = '';
+      body += chunk;
+    });
+    req.on('end', () => {
+      res.writeHead(200);
+      res.end(postHTML);
+    });
+    res.write(`${postHTML}`);
+  } else if (req.url === '/submit') { // Route for submit takes data submitted from form and displays it to this route.
+    res.writeHead(200, { 'Content-Type': 'text/html' });
+    const newData = new URLSearchParams(body);
+    const parsedData = newData.values();
+    const [name, email, text, check] = parsedData;
+    let toText = text;
+    let toPrint = '';
+    if (toText === '') {
+      toText = 'n/a';
+    }
+    if (check === 'on') {
+      toPrint = 'Yes, sign me up for the newsletter';
+    } else {
+      toPrint = 'No, thank you';
+    }
+    res.write(`<p>Name: ${name} </p>`);
```

Comment:
Easy solution for printing the data. I ended up using the key data from the url but I needed to change the html element id's to have the first letter capitalized just for formatting purposes lol. I should have just done it like this